aspnetcore/fundamentals/w3c-logger/index.md

```diff
@@ -0,0 +1,116 @@
+
+[!code-csharp[](samples/6.x/Startup.cs?name=snippet&highlight=3)]
+
+By default, W3CLogger logs common properties such as path, status-code, date, time, and protocol. All information about a single request/response pair is written to the same line.
+
+```
+#Version: 1.0
+#Start-Date: 2021-09-29 22:18:28
+#Fields: date time c-ip s-computername s-ip s-port cs-method cs-uri-stem cs-uri-query sc-status time-taken cs-version cs-host cs(User-Agent) cs(Referer)
+2021-09-29 22:18:28 ::1 DESKTOP-LH3TLTA ::1 5000 GET / - 200 59.9171 HTTP/1.1 localhost:5000 Mozilla/5.0+(Windows+NT+10.0;+WOW64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/93.0.4577.82+Safari/537.36 -
+2021-09-29 22:18:28 ::1 DESKTOP-LH3TLTA ::1 5000 GET / - 200 0.1802 HTTP/1.1 localhost:5000 Mozilla/5.0+(Windows+NT+10.0;+WOW64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/93.0.4577.82+Safari/537.36 -
+2021-09-29 22:18:30 ::1 DESKTOP-LH3TLTA ::1 5000 GET / - 200 0.0966 HTTP/1.1 localhost:5000 Mozilla/5.0+(Windows+NT+10.0;+WOW64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/93.0.4577.82+Safari/537.36 -
+```
+
+## W3C Logger options
+
+To configure the W3CLogger middleware, call `AddW3CLogging` in `ConfigureServices`.
+
+[!code-csharp[](samples/6.x/Startup.cs?name=configureservices)]
+
+### `LoggingFields`
+
+`W3CLoggerOptions.LoggingFields` is an enum flag that configures specific parts of the request and response to log, as well as other information about the connection. `LoggingFields` defaults to include all possible fields except `UserName` and `Cookie`. 
+
+[!code-csharp[](samples/6.x/Startup.cs?name=configureservices&highlight=6)]
+
+| Flag | Description | Value |
+| ---- | ----------- | :---: |
+| None | No logging. | 0x0 |
+| `Date` | The date that the activity occurred. | 0x1 |
+| `Time` |  The date that the activity occurred. | 0x2 |
+| `ClientIpAddress` |  The IP address of the client that accessed the server. | 0x4 |
+| `UserName` |  The name of the authenticated user that accessed the server. | 0x8 |
+| `ServerName` |  The name of the server on which the log entry was generated. | 0x10 |
+| `ServerIpAddress` |  The IP address of the server on which the log entry was generated. | 0x20 |
+| `ServerPort` |  The port number the client is connected to. | 0x40 |
+| `Method` |  Request <xref:Microsoft.AspNetCore.Http.HttpRequest.Method>. | 0x80 |
+| `UriStem` |  Request Path, which includes both the <xref:Microsoft.AspNetCore.Http.HttpRequest.Path> and <xref:Microsoft.AspNetCore.Http.HttpRequest.PathBase>. | 0x100 |
+| `UriQuery` |  Request <xref:Microsoft.AspNetCore.Http.HttpRequest.QueryString>. | 0x200 |
+| `ProtocolStatus` |  Response <xref:Microsoft.AspNetCore.Http.HttpResponse.StatusCode>. | 0x400 |
+| `TimeTaken` |  The duration of time, in milliseconds, that the action consumed. | 0x800 |
+| `ProtocolVersion` |  Request <xref:Microsoft.AspNetCore.Http.HttpRequest.Protocol>. | 0x1000 |
+| `Host` |  Request <xref:Microsoft.AspNetCore.Http.Headers.HeaderNames.Host>. | 0x2000 |
```

Comment:
@Rick-Anderson will this work? The `HeaderNames` stuff I'm referencing are all static strings: https://github.com/dotnet/aspnetcore/blob/3d7a624cc7bfbfeec1843efb8c00b4832e7c5675/src/Http/Headers/src/HeaderNames.cs#L148

Reply:
I guess not:

> Cross reference not found: 'Microsoft.AspNetCore.Http.Headers.HeaderNames.Host'.

Any better way to do this?

Reply:
That's probably cause the API docs haven't been updated for 6.0 yet

Reply:
> That's probably cause the API docs haven't been updated for 6.0 yet

Not all 6.0 have been updated. Code fence it for now. Feel free to open an issue and assign to me (with POST 6.0 label) to add `xref`.

Reply:
What's the formatting that you're suggesting?

Reply:
> What's the formatting that you're suggesting?
code fenced as my suggestions below

```suggestion
| `Host` |  Request `Host` | 0x2000 |
```
or
```suggestion
| `Host` |  Request `Microsoft.AspNetCore.Http.Headers.HeaderNames.Host` | 0x2000 |
```
Or
```suggestion
| `Host` |  Request `Http.Headers.HeaderNames.Host` | 0x2000 |
```

Reply:
https://github.com/dotnet/AspNetCore.Docs/issues/23421